ksql-engine/src/test/java/io/confluent/ksql/physical/PhysicalPlanBuilderTest.java

```diff
@@ -199,9 +208,45 @@ public void shouldHaveKStreamDataSource() {
   }
 
   @Test
-  public void shouldHaveOutputNode() {
+  public void shouldMakeBareQuery() {
     final QueryMetadata queryMetadata = buildPhysicalPlan(simpleSelectFilter);
-    assertThat(queryMetadata.getOutputNode(), instanceOf(KsqlBareOutputNode.class));
+    assertThat(queryMetadata, instanceOf(QueuedQueryMetadata.class));
+  }
+
+  @Test
+  public void shouldMakePersistentQuery() {
+    // Given:
+    givenKafkaTopicExists("test1");
+
+    // When:
+    final QueryMetadata queryMetadata =
+        buildPhysicalPlan("CREATE STREAM FOO AS " + simpleSelectFilter);
+
+    // Then:
+    assertThat(queryMetadata, instanceOf(PersistentQueryMetadata.class));
+  }
+
+  @Test
+  public void shouldBuildMapValuesNodeForTransientQueries() {
+    final QueryMetadata query = buildPhysicalPlan(simpleSelectFilter);
+
+    final TopologyDescription.Processor node = getNodeByName(query, FILTER_MAPVALUES_NODE);
+
+    verifyProcessorNode(node, ImmutableList.of(FILTER_NODE), ImmutableList.of(FOREACH_NODE));
+  }
+
+  private static Processor getNodeByName(final QueryMetadata query, final String nodeName) {
```

Comment:
nit: move to bottom of file. Also, no given/when/then pattern for these new tests?

Reply:
Terribly sorry. They were a cut & paste job from another file.  Sorted :D